crates/router/src/core/payment_link.rs

```diff
@@ -197,6 +206,35 @@ fn validate_sdk_requirements(
     Ok((pub_key, currency, client_secret))
 }
 
+pub async fn list_payment_link(
+    state: AppState,
+    merchant: domain::MerchantAccount,
+    constraints: api_models::payments::PaymentLinkListConstraints,
+) -> RouterResponse<Vec<api_models::payments::RetrievePaymentLinkResponse>> {
+    // helpers::validate_payment_link_list_request(&constraints)?;
+    let db = state.store.as_ref();
+    let payment_link = db
+        .find_payment_link_by_merchant_id(&merchant.merchant_id, constraints)
```

Comment:
Can we rename `find_payment_link_by_merchant_id` to denote that we are getting Vec of links and not a single payment_link